src/test/java/edu/ucsb/ucsbcslas/controllers/OnlineOfficeHourControllerTests.java

```diff
@@ -251,4 +267,158 @@ public void testDeleteOfficeHour_courseNotFound() throws Exception {
+        );
+        MockMvc mockMvc = MockMvcBuilders.webAppContextSetup(webApplicationContext).build();
+        mockMvc
+            .perform(multipart("/api/admin/officehours/upload").file(mockFile)
+                .characterEncoding("utf-8")
+                .header(HttpHeaders.AUTHORIZATION, "Bearer " + userToken()))
+            .andExpect(status().isOk())
+            .andReturn();
+        verify(mockOnlineOfficeHoursRepository, times(1)).save(any());
+    }
+
+
+    @Test
+    public void testUploadFile_prexist() throws Exception{
+        List<OnlineOfficeHours> expectedOfficeHours = new ArrayList<OnlineOfficeHours>();
+        List<TutorAssignment> TAL = new ArrayList<> ();
+        Tutor t = new Tutor(1L, "String firstName", "String lastName", "email@ucsb.edu");
+        Course c = new Course(1L, "String name", "F20", "String instructorFirstName", "String instructorLastName", "insEmail@ucsb.edu");
+        Optional<Tutor> OptTutor = Optional.of(t);
+        TutorAssignment tutorAss = new TutorAssignment(1L, c, t, "String assignmentType");
+        TAL.add(tutorAss);
+        OnlineOfficeHours oh = new OnlineOfficeHours(1L, tutorAss,"Wednesday", "8:00", "10:00", "link", "notes");
+        expectedOfficeHours.add(oh);
+        
+        String fcontent = "\"String name\",\"F20\",\"String instructorFirstName\",\"String instructorLastName\",\"insEmail@ucsb.edu\",\"String firstName\",\"String lastName\",\"String email\",\"String assignmentType\",\"Wednesday\",\"8:00\",\"10:00\",\"link\",\"notes\"";
+        when(mockAuthControllerAdvice.getIsAdmin(anyString())).thenReturn(true);
+        when(mockCourseRepository.findByNameAndQuarter(any(String.class), any(String.class))).thenReturn(c);
+        when(mockTutorRepository.findByEmail(any(String.class))).thenReturn(OptTutor);
+        when(mockTutorAssignmentRepository.findAllByCourse(any(Course.class))).thenReturn(TAL);
+        when(mockTutorAssignmentRepository.findAllByTutor(any(Tutor.class))).thenReturn(TAL);
+        when(mockOnlineOfficeHoursRepository.findAllByTutorAssignment(any(TutorAssignment.class))).thenReturn(expectedOfficeHours);
+        when(mockOnlineOfficeHoursRepository.save(any(OnlineOfficeHours.class))).thenReturn(oh);
+        MockMultipartFile mockFile = new MockMultipartFile(
+            "csv",
+            "test.csv",
+            MediaType.TEXT_PLAIN_VALUE,
+            fcontent.getBytes("utf-8")
+        );
+        MockMvc mockMvc = MockMvcBuilders.webAppContextSetup(webApplicationContext).build();
+        mockMvc
+            .perform(multipart("/api/admin/officehours/upload").file(mockFile)
+                .characterEncoding("utf-8")
+                .header(HttpHeaders.AUTHORIZATION, "Bearer " + userToken()))
+            .andExpect(status().isOk())
+            .andReturn();
+        verify(mockOnlineOfficeHoursRepository, times(0)).save(any());
+    }
+
+    @Test
+    public void testUploadFile_3OH() throws Exception{
```

Comment:
Small nitpick, but this test name could be more informative. It's hard to immediately tell what is being tested when reading it

Reply:
The test is used to test the function to checking whetehr new OH has the same `dayOfWeek`, `startTime` and `endTime` with the preexisted ones. The test name is changed to `testUploadFile_DifferentDayofWeekStartandEndtime()`